runtime/jcl/common/jdk_jfr_internal_JVM_jdk17andUp.cpp

```diff
@@ -84,8 +86,29 @@ Java_jdk_jfr_internal_JVM_getChunkStartNanos(JNIEnv *env, jobject obj)
 jlong JNICALL
 Java_jdk_jfr_internal_JVM_getTypeId__Ljava_lang_String_2(JNIEnv *env, jobject obj, jstring name)
 {
-	// TODO: implementation
-	return 0;
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	jlong result = -1;
+	char buf[STRING_BUFFER_LENGTH];
```

Comment:
Suggest removing the definition of `STRING_BUFFER_LENGTH` and hard-coding 128 here; line 99 should use `sizeof(buf)` instead of the macro.

Reply:
I don't mind the constant definition for the size, but the use of sizeof the buffer should be our standard either way.